### ISSUE TITLE
F.max_pool1d (and probably min_pool1d) should support integer dtypes

### ISSUE THREAD

**Issue by vadimkantorov** [pytorch/pytorch#33086]
max_pool and min_pool are dilation and erosion operations which can be convenient in postprocessing, so binary masks input can happen in practice

**Comment by fmassa** [comment]
I agree that it would be convenient to support more operations on integer types, like interpolate.

**Comment by vadimkantorov** [comment]
This can also be used for simple connected component propagation: https://github.com/kornia/kornia/blob/f1eae7b0f5d67490a7be962896ee59ec73eb779a/kornia/contrib/connected_components.py

**Comment by vadimkantorov** [comment]
Ideally, it should work without any casts, for uint8/bool it should be quite a huge memory and speed boost
